capa/features/extractors/viv/syscall.py

```diff
@@ -0,0 +1,1359 @@
+        return None
+    return name
+
+
+def add_library_function(vw, bb: BBHandle, get_library_function_name_: Callable) -> None:
+    name = get_library_function_name_(vw, bb)
+    if not name:
+        return
+    try:
+        add = AbsoluteVirtualAddress(bb.inner.instructions[0].va)
+        if name == "exit":  # dont hook exit()
+            return
+        viv_utils.flirt.make_library_function(vw, add)
+        viv_utils.set_function_name(vw, add, name)
+    # not a function addresss
+    except Exception:
+        return
+
+
+def is_static(buf: bytes) -> bool:
+    byteorder = "little" if buf[0x05] == 1 else "big"
+    arch = buf[0x04]
+
+    # 32-bit format
+    if arch == 1:
+        e_shoff = int.from_bytes(buf[0x20:0x24], byteorder)
+        e_shentsize = int.from_bytes(buf[0x2E:0x30], byteorder)
+        e_shstr = e_shoff + e_shentsize * int.from_bytes(buf[0x32:0x34], byteorder)
+        str_sh_offset = int.from_bytes(buf[e_shstr + 0x10 : e_shstr + 0x14], byteorder)
+        str_sh_size = int.from_bytes(buf[e_shstr + 0x14 : e_shstr + 0x20], byteorder)
+    # 64-bit format
+    else:
+        e_shoff = int.from_bytes(buf[0x28:0x30], byteorder)
+        e_shentsize = int.from_bytes(buf[0x3A:0x3C], byteorder)
+        e_shstr = e_shoff + e_shentsize * int.from_bytes(buf[0x3E:0x40], byteorder)
+        str_sh_offset = int.from_bytes(buf[e_shstr + 0x18 : e_shstr + 0x20], byteorder)
+        str_sh_size = int.from_bytes(buf[e_shstr + 0x20 : e_shstr + 0x28], byteorder)
+
+    strings = buf[str_sh_offset : str_sh_offset + str_sh_size].split(b"\x00")
+    return b".dynamic" not in strings
+
+
+def resolve_syscall_functions(extractor: FeatureExtractor):
+    with open(extractor.path, "rb") as f:
+        buf = f.read()
+    # only resolve syscall if the file is staticaly linked
+    if not is_static(buf):
+        return
+    import capa.features.extractors.common
+
+    arch = list(capa.features.extractors.common.extract_arch(buf))[0][0].value
+    del buf  # free memory
+
+    functions = list(extractor.get_functions())
+    if arch in (ARCH_I386, ARCH_AMD64):
+        for function in functions:
+            for bb in extractor.get_basic_blocks(function):
+                insn = list(extractor.get_instructions(function, bb))
+
+                vw = extractor.vw
+                mnem_list = [i.inner.mnem for i in insn]
+                if not (mnem_list.count("int" if vw.psize == 4 else "syscall") == 1):
+                    continue
+                add_library_function(extractor.vw, bb, get_library_function_name)
+
+    elif arch == ARCH_ARM:
+        for function in functions:
+            for bb in extractor.get_basic_blocks(function):
+                insn = list(extractor.get_instructions(function, bb))
+                mnem_list = [i.inner.mnem for i in insn]
+                if mnem_list.count("svc") == 1:  # bx lr
+                    add_library_function(extractor.vw, bb, get_library_function_name_arm)
+
+
+Sys_x64 = {
```

Comment:
if we want to do this for the other backends as well, we can move the mappings to a file one directory up

Reply:
Do you want us to create this file now ?

Reply:
if you don't mind, please... and thanks!